scripts/viz/feature_activation_plots.R

```diff
@@ -0,0 +1,132 @@
+# Pan-Cancer Variational Autoencoder
+# Gregory Way 2017
+#
+# Visualize VAE feature activation
+#
+# 1) Scatter plots of specific covariate information by encodings
+# 2) Heatmap of full encoded features by all samples
+
+library(readr)
+library(dplyr)
+library(ggplot2)
+
+# 1) Scatter plots of encoding specific sample activation
+# Load the combined data frame where columns are different data and algorithms
+vae_clinical_file <- file.path("data", "tybalt_features_with_clinical.tsv")
+combined_df <- readr::read_tsv(vae_clinical_file)
+
+# Load official colors
+tcga_colors <- readr::read_tsv(file.path("data", "tcga_colors.tsv"))
+
+tcga_colors <- tcga_colors[order(tcga_colors$`Study Abbreviation`), ]
+match_colors <- match(combined_df$acronym, tcga_colors$`Study Abbreviation`)
+combined_df$colors <- tcga_colors$`Hex Colors`[match_colors]
+
+palette_order <- c("BRCA", "PRAD", "TGCT", "KICH", "KIRP", "KIRC", "BLCA",
+                   "OV", "UCS", "CESC", "UCEC", "THCA", "PCPG", "ACC", "SKCM",
+                   "UVM", "HNSC", "SARC", "ESCA", "STAD", "COAD", "READ",
+                   "CHOL", "PAAD", "LIHC", "DLBC", "MESO", "LUSC", "LUAD",
+                   "GBM", "LGG", "LAML", "THYM", NA)
+
+plot_vae <- function(x, y, covariate, legend_show = TRUE) {
+  # Helper function to plot given coordinates and color by specific covariates
+  x_coord <- paste(x)
+  y_coord <- paste(y)
+  color_ <- covariate
+
+  p <- ggplot(combined_df,
+              aes_string(x = combined_df[[x_coord]],
+                         y = combined_df[[y_coord]],
+                         color = color_)) + 
+    geom_point() +
+    xlab(paste("encoding", x_coord)) +
+    ylab(paste("encoding", y_coord)) +
+    theme_bw() +
+    theme(text = element_text(size = 20))
+  
+  if (color_ == "acronym") {
+    p <- p + scale_colour_manual(limits = tcga_colors$`Study Abbreviation`,
+                                 values = tcga_colors$`Hex Colors`,
+                                 na.value = "black",
+                                 breaks = palette_order)
+  }
+
+  if (!legend_show) {
+    p <- p + theme(legend.position = "none")
+  }
+  print(p)
+}
+
+gender_encodings <- file.path("figures", "gender_encodings.pdf")
+gender_encodings_legend <- file.path("figures", "gender_encodings_legend.pdf")
+sample_type_file <- file.path("figures", "sample_type_encodings.pdf")
+sample_legend_file <- file.path("figures", "sample_type_encodings_legend.pdf")
+
+# Output plots with and without legend
+plot_vae(x = 82, y = 85, covariate = "gender", legend_show = FALSE)
+ggsave(gender_encodings, width = 6, height = 5)
+
+plot_vae(x = 82, y = 85, covariate = "gender", legend_show = TRUE)
+ggsave(gender_encodings_legend, width = 6, height = 5)
+
+# Subset new dataframe for visualizing Melanoma activation
+met_df <- combined_df %>%
+  dplyr::select("53", "66", "sample_type", "acronym") %>%
```

Comment:
Are `53` and `66` the nodes you're interested in here? Can you put a comment to that effect?

Reply:
:+1: